### PR TITLE
Add flexible filename and extension to phonon new_plotter

### PR DESCRIPTION
**Pull request by QuantumChemist**
Finally had some time to do this.

I added a way to set a flexible filename and extension for the phonon band structure and DOS new_plotter object.

This PR also makes this PR https://github.com/materialsproject/atomate2/pull/624 obsolete.

### CODE REVIEW

**Comment by QuantumChemist** [comment]
according to @janosh's suggestion here https://github.com/materialsproject/atomate2/pull/624#discussion_r1406637380 :)

**Comment by QuantumChemist** [comment]
@janosh I hope it's ok if I also ask whether this PR is ok like that?

File: src/atomate2/common/schemas/phonons.py [hunk 1]
```diff
@@ -359,9 +359,9 @@ def from_forces_born(
             filename_band_yaml, labels_dict=kpath_dict, has_nac=born is not None
         )
         new_plotter = PhononBSPlotter(bs=bs_symm_line)
-
         new_plotter.save_plot(
-            "phonon_band_structure.eps", units=kwargs.get("units", "THz")
+            filename=kwargs.get("filename_BS", "phonon_band_structure.eps"),
```
Comment (janosh) [inline src/atomate2/common/schemas/phonons.py:363]: let's stick with snake_case here: `filename_bs` and `filename_dos`

File: tests/common/schemas/test_phonons.py [hunk 2]
```diff
@@ -58,3 +59,31 @@ def test_phonon_bs_dos_doc():
...
+    phojob = PhononMaker(
+        bulk_relax_maker=CHGNetRelaxMaker(steps=5),
+        phonon_displacement_maker=CHGNetStaticMaker(),
+        static_energy_maker=CHGNetStaticMaker(),
+        store_force_constants=False,
+        generate_frequencies_eigenvectors_kwargs={
+            "filename_BS": "phonon_BS_test.png",
+            "filename_DOS": "phonon_DOS_test.pdf",
+        },
+    ).make(structure=si_structure)
```
Comment (janosh) [inline tests/common/schemas/test_phonons.py:79]: can we avoid running the workflow given we're only testing the `from_forces_born` method?

**Review by janosh — CHANGES_REQUESTED** [review]
no problem. similar comment re keeping test time low as in #791

File: tests/common/schemas/test_phonons.py (hunk 2 above)
Comment (QuantumChemist) [reply tests/common/schemas/test_phonons.py]: I have moved the tests now to `tests/forcefields/flows/test_phonon.py`.  It's not a schemas test anymore, then, but that way, we can avoid unnecessary run time.

**Comment by QuantumChemist** [comment]
@janosh I implemented the requested changes and hope it's ok that way!

**Review by janosh — APPROVED** [review]
thanks @QuantumChemist, looks great!

**Pull request merged**